hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metrics/Metrics.java

```diff
@@ -97,6 +110,14 @@ public static synchronized void shutdown() {
     initialized = false;
   }
 
+  public static synchronized void flush() {
+    if (!Metrics.initialized) {
+      return;
```

Comment:
What should be the right expectation if it's not initialized ? Simply returning gives a false impression to the client that it's flushed.

Reply:
This is inline with the shutdown() and init() functions which are fail-safe. Unless metrics are initialized there is no effect of calling these.